AEPCore/Sources/eventhub/history/EventHistoryDatabase.swift

```diff
@@ -0,0 +1,232 @@
+        self.dispatchQueue = dispatchQueue
+        guard createTable() else {
+            Log.warning(label: LOG_PREFIX, "Failed to initialize Event History Database.")
+            return nil
+        }
+        guard let dbConnection = connect() else {
+            Log.warning(label: LOG_PREFIX, "Failed to connect to Event History Database.")
+            return nil
+        }
+        self.connection = dbConnection
+    }
+
+    deinit {
+        if let dbConnection = connection {
+            disconnect(database: dbConnection)
+        }
+    }
+
+    /// Inserts a record in the EventHistory database.
+    ///
+    /// Fails if a connection to the database cannot be established, and calls the `handler` with a value of `false`.
+    ///
+    /// The current timestamp will always be used on this insert.
+    ///
+    /// - Parameters:
+    ///   - hash: the hashed value representing an event.
+    ///   - handler: called with the `Bool` result of the insert statement.
+    func insert(hash: UInt32, handler: ((Bool) -> Void)? = nil) {
+        dispatchQueue.async {
+            // first verify we can get a connection handle
+            guard let connection = self.connection else {
+                handler?(false)
+                return
+            }
+
+            let insertStatement = """
+            INSERT INTO \(self.tableName) (\(self.columnHash), \(self.columnTimestamp))
+            VALUES (\(hash), \(Date().millisecondsSince1970))
+            """
+
+            let result = SQLiteWrapper.execute(database: connection, sql: insertStatement)
+            handler?(result)
+        }
+    }
+
+    /// Queries the event history database to search for existence of an event.
+    ///
+    /// This method will count all records in the event history database that match the provided `hash` and are within
+    /// the bounds of the provided `from` and `to` date.
+    ///
+    /// If no `from` date is provided, the search will use the beginning of event history
+    /// as the lower bounds of the date range.
+    ///
+    /// If no `to` date is provided, the search will use `now` as the upper bounds
+    /// of the date range.
+    ///
+    /// The `handler` will be called with an `EventHistoryResult`.
+    ///
+    /// If no database connection is available, the handler will be called with a count of 0.
+    /// If there are no matching records, the handler will be called with count of 0.
+    ///
+    /// - Parameters:
+    ///   - hash: the 32-bit FNV-1a hashed representation of an Event's data.
+    ///   - from: represents the lower bounds of the date range to use when searching for the hash
+    ///   - to: represents the upper bounds of the date range to use when searching for the hash
+    ///   - handler: a callback which will contain `EventHistoryResult` representing matching events
+    func select(hash: UInt32, from: Date? = nil, to: Date? = nil, handler: @escaping (EventHistoryResult) -> Void) {
+        dispatchQueue.sync {
```

Comment:
the `dispatchQueue` should handle any potential race conditions.  `select` is an async method, but we are accessing the `dispatchQueue` synchronously because the caller is waiting for a result.